src/Simplex/Chat/Store.hs

```diff
@@ -1033,11 +1033,13 @@ deleteContactRequest db userId contactRequestId = do
     (userId, userId, contactRequestId)
   DB.execute db "DELETE FROM contact_requests WHERE user_id = ? AND contact_request_id = ?" (userId, contactRequestId)
 
-createAcceptedContact :: DB.Connection -> UserId -> ConnId -> ContactName -> ProfileId -> Profile -> Int64 -> Maybe XContactId -> Maybe Profile -> IO Contact
-createAcceptedContact db userId agentConnId localDisplayName profileId profile userContactLinkId xContactId incognitoProfile = do
+createAcceptedContact :: DB.Connection -> UserId -> ConnId -> ContactName -> ProfileId -> Profile -> Int64 -> Maybe XContactId -> Maybe Profile -> Maybe ProfileId -> IO Contact
+createAcceptedContact db userId agentConnId localDisplayName profileId profile userContactLinkId xContactId incognitoProfile existingIncognitoProfileId = do
   DB.execute db "DELETE FROM contact_requests WHERE user_id = ? AND local_display_name = ?" (userId, localDisplayName)
   createdAt <- getCurrentTime
-  customUserProfileId <- createIncognitoProfile_ db userId createdAt incognitoProfile
+  customUserProfileId <- case existingIncognitoProfileId of
+    Just _ -> pure existingIncognitoProfileId
+    Nothing -> createIncognitoProfile_ db userId createdAt incognitoProfile
```

Comment:
```suggestion
  customUserProfileId <- (existingIncognitoProfileId <|>) <$> createIncognitoProfile_ db userId createdAt incognitoProfile
```

Reply:
something like that can be possible... This is incorrect, because it would always call createIncognitoProfile_

Reply:
maybe:

```
runMaybeT $ pure existingIncognitoProfileId <|> MaybeT (createIncognitoProfile_ db userId createdAt incognitoProfile)
```